lib/segment/src/vector_storage/query_scorer/mod.rs

```diff
@@ -58,3 +63,45 @@ fn score_multi<T: PrimitiveVectorElement, TMetric: Metric<T>>(
         }
     }
 }
+
+/// Check if ids are rather contiguous to enable further optimizations
+/// The threshold is 20% of sparsity
+pub fn check_ids_rather_contiguous(ids: &[PointOffsetType]) -> bool {
+    if ids.len() < 2 {
+        return false;
+    }
+    // check if sorted
+    if ids.windows(2).any(|w| w[0] >= w[1]) {
```

Comment:
Check for sorted is `O(N)`, but for the same price we can just find min & max and don't really care if it is strictly sorted